ios/objective-c/AmpliObjectiveCSampleApp/AmpliObjectiveCSampleAppTests/AmpliTests.m

```diff
@@ -20,14 +21,34 @@ - (void)setUp {
     [_ampli load:[LoadOptions builderBlock:^(LoadOptionsBuilder *b) {
         b.apiKey = @"test-api-key";
     }]];
+    _ampli.client.eventUploadThreshold = 1;
+    _hasMiddlewareRun = NO;
+    [_ampli flush];
+}
+
+- (void) waitForMiddlewareToRun {
```

Comment:
I would use deterministic `XCTestExpectation` to wait for middleware completion.
Additionally: new Amplitude instance for each test with new fresh storage (maybe fake/in-memory)
It looks like Swift unit tests in CI fail for similar reason - the example app is started with empty api key.

Reply:
Thanks, I tried `XCTestExpectation` but it caused the tests to run multiple instances and threading issues. I am going to dig in more today.

Reply:
Got it working with `XCTestExpectation`. Threading issues were resolved by using new instances of Ampli and Amplitude for each test.